test/e2e/specs/specs-playwright/wp-widgets__spec.ts

```diff
@@ -1,8 +1,15 @@
-import { DataHelper, LoginFlow, SidebarComponent, setupHooks } from '@automattic/calypso-e2e';
+import {
+	BrowserHelper,
+	DataHelper,
+	LoginFlow,
+	SidebarComponent,
+	setupHooks,
+} from '@automattic/calypso-e2e';
 import { Page } from 'playwright';
 
-const user =
-	process.env.GUTENBERG_EDGE === 'true' ? 'gutenbergSimpleSiteEdgeUser' : 'gutenbergSimpleSiteUser';
+const user = BrowserHelper.targetGutenbergEdge()
+	? 'gutenbergSimpleSiteEdgeUser'
+	: 'gutenbergSimpleSiteUser';
```

Comment:
I suppose we want to keep the actual ternary logic here instead of moving it inside `targetGutenbergEdge` method itself? 

Would it make sense to set up a separate job for `edge` site testing instead of having a branching condition in the test suite itself? A possible scenario for confusion may be as follows: without digging into how the environment is set up, a viewer may not know whether this suite is being run with a normal or edge user.

@fullofcaffeine what do you think about separating `edge` jobs into a different task in TeamCity?

Reply:
> I suppose we want to keep the actual ternary logic here instead of moving it inside targetGutenbergEdge method itself?

I considered this, and don't mind making the switch if you'd like! I opted this way because 1) I thought `targetGutenbergEdge` might be used for more than just users, and 2) most scripts list their user in them, and so I didn't want to bury this user in the package.

> Would it make sense to set up a separate job for edge site testing instead of having a branching condition in the test suite itself?

Could you flesh this out a little bit more? I'm not sure I follow, sorry! I definitely like the idea of separate TC jobs, but I think we'd still need some way to identify that's what the script should target, so I think we'd still need a conditional, right?

Reply:
> most scripts list their user in them, and so I didn't want to bury this user in the package.

This is along the lines I was thinking as well.

>> Would it make sense to set up a separate job for edge site testing instead of having a branching condition in the test suite itself?
> Could you flesh this out a little bit more?

Sure! Hear me out, and let me know if my observation makes any sense or is warranted at all.

I do not think a single build configuration should run tests that may produce different results depending on the environment variable that is set.

A hypothetical scenario - 

- most of the time, tests on WPCOM is run with `gutenbergSimpleSiteUser`.
- these tests pass with 99% passing rate.
- at some interval, `gutenbergSimpleSiteEdgeUser` is triggered manually.
- these tests have a 50% passing rate.

It can result in the following hypothetical outcomes:

- lower passing rate when testing with `gutenbergSimplesiteEdgeUser` will skew overall reliability of the configuration.
- those looking at configuration may not know which runs were with normal and which were with edge user.

I think the second point is key. We have two separate build configurations for the viewport/devices that are tested (mobile/desktop). These are separated because altering the target device changes the baseline environment. If my understanding is correct, changing the envvar to trigger `edge` testing also changes the baseline => therefore we should strive to keep these tasks separate.

I hope I articulated it a bit better than before - I'd be happy to go into more detail if not!

Reply:
Oh for sure! Thanks for fleshing that out more. I 💯  agree with you. Makes reporting and tracking metrics over time much much cleaner!

Do you think there's anything I need to add to this spec to support that? Otherwise I can cut a PR in the future splitting the Gutenberg edge testing into two different TeamCity builds!

Reply:
I was thinking of having @fullofcaffeine chime in on whether we should separate edge and normal Gutenberg testing. With what I've said in the previous comment though, let's not add complication for the time being and go with keeping the 'status quo'.  We can always split things up later when warranted :)